bluesky/spec_api.py

```diff
@@ -71,15 +73,61 @@ def setup_plot(motors):
         return LivePlot(y_key, fig=fig)
 
 
-def setup_peakstats(motors):
+def setup_peakstats(motors, dets, gs):
     "Set up peakstats"
-    key = first_key_heuristic(list(motors)[0])
-    ps = PeakStats(key, gs.MASTER_DET_FIELD, edge_count=3)
+    motor = list(motors)[0]
+    key = first_key_heuristic(motor)
+    ps = PeakStats(key, gs.MASTER_DET_FIELD, **gs.PS_CONFIG)
     gs.PS = ps
+    ps.motor = motor
     return ps
 
 
-### Counts (p. 140) ###
+def setup_livetable(motors, dets, gs):
+    return LiveTable(motors + [gs.PLOT_Y] + gs.TABLE_COLS)
+
+
+def setup_liveraster(motors, dets, gs, *, shape, extent):
+    if len(motors) != 2:
+        return None
+    ylab, xlab = [first_key_heuristic(m) for m in motors]
+    raster = LiveRaster(shape, gs.MASTER_DET_FIELD, xlabel=xlab,
+                        ylabel=ylab, extent=extent, clim=[0, 1])
+    return raster
+
+
+def _construct_subs(plan_name, motors, dets, **kwargs):
```

Comment:
I think this should be public, since I should be able to write other `gs`-abiding scans if I so choose.